druntime/src/core/builtins.d

```diff
@@ -51,3 +51,32 @@ version (LDC)
 
 /// Writes `s` to `stderr` during CTFE (does nothing at runtime).
 void __ctfeWrite(scope const(char)[] s) @nogc @safe pure nothrow {}
+
+version (GNU)
+{
+    alias expect = __builtin_expect;
+    alias trap = __builtin_trap;
+}
+else version (LDC)
+{
+    alias expect = llvm_expect;
+    debug
+        alias trap = llvm_debugtrap;
+    else
+        alias trap = llvm_trap;
+}
+else version (DigitalMars)
+{
+    pragma(inline, true)
+    T expect(T)(T val, T expected) if (__traits(isIntegral, T))
+    {
+        return val;
+    }
+
+    pragma(inline, true)
+    noreturn trap() { assert(0); }
+}
+
+/// Provide static branch hints
+pragma(inline, true) bool likely(bool b)   { return expect(b, true);  }
+pragma(inline, true) bool unlikely(bool b) { return expect(b, false); }
```

Comment:
Public documentation should have `Params` and `Returns` sections. 

If there's one documentation comment for both:
```suggestion
/// ditto
pragma(inline, true) bool unlikely(bool b) { return expect(b, false); }
```

Reply:
done